cloudsdk/src/main/java/io/particle/android/sdk/cloud/ParticleDevice.java

```diff
@@ -406,6 +406,31 @@ public void executeFlashingChange() throws RetrofitError {
         });
     }
 
+    @WorkerThread
+    public void flashCodeFile(final File file) throws ParticleCloudException {
+        performFlashingChange(new FlashingChange() {
+            @Override
+            public void executeFlashingChange() throws RetrofitError {
+                mainApi.flashFile(deviceState.deviceId,
+                        new TypedFile("multipart/form-data", file));
+            }
+        });
+    }
+
+
+    @WorkerThread
```

Comment:
I couldn't get your first question.

As for the second, I should have used `multipart/form-data`. Fixed in the current commit